app/scripts/lib/multichain-api/wallet-requestPermissions.test.js

```diff
@@ -151,283 +175,454 @@ describe('requestPermissionsHandler', () => {
-      process.env.BARAD_DUR = 1;
-    });
+  it('does not request permissions when only permittedChains is provided in params', async () => {
+    const { handler, requestPermissionsForOrigin } = createMockedHandler();
 
-    it('does not update or grant a CAIP-25 endowment type permission if `eth_accounts` permissions were not granted', async () => {
-      const {
-        handler,
-        requestPermissionsForOrigin,
-        updateCaveat,
-        grantPermissions,
-      } = createMockedHandler();
-      requestPermissionsForOrigin.mockResolvedValue([{}]);
-
-      await handler(baseRequest);
-      expect(updateCaveat).not.toHaveBeenCalled();
-      expect(grantPermissions).not.toHaveBeenCalled();
+    await handler({
+      ...getBaseRequest(),
+      params: [
+        {
+          [PermissionNames.permittedChains]: {},
+        },
+      ],
     });
+    expect(requestPermissionsForOrigin).not.toHaveBeenCalled();
+  });
 
-    it('returns the unmodified granted permissions if eth_accounts was not granted', async () => {
-      const { handler, requestPermissionsForOrigin, response } =
-        createMockedHandler();
-      requestPermissionsForOrigin.mockResolvedValue([
+  it('does not request permissions when both eth_accounts and permittedChains are provided in params', async () => {
+    const { handler, requestPermissionsForOrigin } = createMockedHandler();
+
+    await handler({
+      ...getBaseRequest(),
+      params: [
         {
-          otherPermission: {
-            id: '3',
-            parentCapability: 'otherPermission',
+          [RestrictedMethods.eth_accounts]: {},
+          [PermissionNames.permittedChains]: {
             caveats: [
               {
-                value: {
-                  foo: 'bar',
-                },
+                type: CaveatTypes.restrictNetworkSwitching,
+                value: ['0x64'],
               },
             ],
           },
         },
-      ]);
+      ],
+    });
+    expect(requestPermissionsForOrigin).not.toHaveBeenCalled();
+  });
 
-      await handler(baseRequest);
-      expect(response.result).toStrictEqual([
+  it('requests empty permissions when only CAIP-25 permission is provided in params', async () => {
+    const { handler, requestPermissionsForOrigin } = createMockedHandler();
+
+    await handler({
+      ...getBaseRequest(),
+      params: [
         {
-          id: '3',
-          parentCapability: 'otherPermission',
-          caveats: [
-            {
-              value: {
-                foo: 'bar',
-              },
-            },
-          ],
+          [Caip25EndowmentPermissionName]: {},
         },
-      ]);
+      ],
     });
+    expect(requestPermissionsForOrigin).toHaveBeenCalledWith({});
+  });
 
-    it('gets permission for the origin', async () => {
-      const { handler, getPermissionsForOrigin } = createMockedHandler();
+  it('requests empty permissions when no permissions are provided in params', async () => {
```

Comment:
also an error scenario? Seems like it is today?

<img width="533" alt="Screenshot 2024-08-30 at 3 10 21 PM" src="https://github.com/user-attachments/assets/31ddcfa6-775c-407a-bfc7-25f6c040909e">

Reply:
params should include an empty object `[{}]`